api/v1beta1/clusterclass_types.go

```diff
@@ -273,7 +273,8 @@ type PatchSelectorMatch struct {
 // in specific MachineDeploymentClasses in .spec.workers.machineDeployments.
 type PatchSelectorMatchMachineDeploymentClass struct {
 	// Names selects templates by class names.
-	Names []string `json:"names"`
+	// +optional
+	Names []string `json:"names,omitempty"`
```

Comment:
While this fix is fine, the PatchSelectorMatchMachineDeploymentClass field `MachineDeploymentClass` should be a pointer given that's optional and omitempty

Reply:
As discussed will be fixed in https://github.com/kubernetes-sigs/cluster-api/pull/5781